docs/book/src/developer/architecture/controllers/machine-pool.md

```diff
@@ -0,0 +1,114 @@
+# Machine  Controller
+
+![](../../../images/cluster-admission-machinepool-controller.png)
+
+The Machine controller's main responsibilities are:
+
+* Setting an OwnerReference on:
+    * Each MachinePool object to the Cluster object.
+    * The associated BootstrapConfig object.
+    * The associated InfrastructureMachinePool object.
+* Copy data from `BootstrapConfig.Status.DataSecretName` to `MachinePool.Spec.Template.Spec.Bootstrap.DataSecretName` if
+`MachinePool.Spec.Template.Spec.Bootstrap.DataSecretName` is empty.
+* Setting NodeRefs to be able to associate instances and kubernetes nodes.
```

Comment:
Are these two the same? Just wanted to clarify where the NodeRefs are being set and what instances were we referring to.
```suggestion
* Setting NodeRefs on MachinePool instances to be able to associate them with kubernetes nodes.
```